packages/api/src/platforms/vtex/resolvers/productGroup.ts

```diff
@@ -6,4 +6,14 @@ export const StoreProductGroup: Record<string, Resolver<Product>> = {
   hasVariant: (root) => root.skus.map((sku) => enhanceSku(sku, root)),
   productGroupID: ({ product }) => product,
   name: ({ name }) => name,
+  additionalProperty: ({ textAttributes = [], productSpecifications = [] }) => {
+    const specs = new Set(productSpecifications)
+
+    return textAttributes
+      .filter((attribute) => specs.has(attribute.labelKey))
+      .map((attribute) => ({
+        name: attribute.labelKey,
+        value: attribute.labelValue,
+      }))
```

Comment:
I don't know enough about VTEX's specification fields to understand this logic. Could you explain why we get the `additionalProperty` from the `textAttributes` and filter using the `productSpecifications`?

Reply:
Ok, so this is tricky, thanks for asking.

We have two kind of additional properties. We have product and sku additional properties. However, VTEX API returns both kind of mixed. This filter serves to pick product only properties.